### ISSUE TITLE
Ensure parameters are always read on subsequent solves

### ISSUE THREAD

**Issue by albertium** [cvxpy/cvxpy#786]
**Describe the bug**
ECOS seems to produce the same results on problem with cvxpy.log if not rebuilding the problem. 

**To Reproduce**
import cvxpy as cp
import numpy as np

np.random.seed(1234)
w = cp.Variable(2)
mat = cp.Parameter((2, 2), PSD=True)
loss = cp.quad_form(w, mat) - cp.sum(cp.log(w))
problem = cp.Problem(cp.Minimize(loss), [w >= 0])

\# first run
data = np.random.normal(size=(10, 2))
cov = np.matmul(data.T, data)
mat.value = cov
problem.solve()
print(w.value)

\# second run
data = np.random.normal(size=(10, 2))
cov = np.matmul(data.T, data)
mat.value = cov
\# problem = cp.Problem(cp.Minimize(loss), [w >= 0])
problem.solve()
print(w.value)

**Expected behavior**
Would expect that the outputs are different since "cov" is different. On the other hand, if rebuilding cp.Problem under "second run", the result will be different as expected.

**Output**
Without rebuilding the problem:
[0.34270986 0.24460691]
[0.34270986 0.24460691]

Rebuilding the problem:
[0.34270986 0.24460691]
[0.29654106 0.18872048]

**Version**
 - OS: Windows
 - CVXPY Version:1.0.24

Thanks!

**Comment by SteveDiamond** [comment]
Fixed in v1.1.
